src/scheduler/job_info.c

```diff
@@ -2255,6 +2263,14 @@ create_resresv_sets_resdef(status *policy, server_info *sinfo) {
 		defs[i++] = getallres(RES_PREEMPT_TARGETS);
 	defs[i] = NULL;
 
+	for(cur_res = limres; cur_res != NULL; cur_res = cur_res->next) {
+		/* There may be overlap between resdef_to_check and limres */
+		if (!resdef_exists_in_array(defs, cur_res->def)) {
+			defs[i++] = cur_res->def;
+			defs[i] = NULL;
```

Comment:
you can probably remove this null assignment out of the for loop and even remove it from line 2264 to do it just once

Reply:
It's needed for resdef_exists_in_array().  It requires a NULL terminated array.  Every time I add something to the array, I need to terminate it again.